frontend/public/components/utils/resource-log.tsx

```diff
@@ -1,17 +1,19 @@
 import {
-  useState,
-  useEffect,
-  useRef,
+  MouseEventHandler,
+  ReactNode,
+  Ref,
   useCallback,
   useContext,
-  ReactNode,
+  useEffect,
   useMemo,
-  Ref,
+  useRef,
+  useState,
 } from 'react';
 import { useSelector } from 'react-redux';
 import { Base64 } from 'js-base64';
 import * as _ from 'lodash-es';
 import { Trans, useTranslation } from 'react-i18next';
+import { detect } from 'chardet';
```

Comment:
Wouldn't it be better to introduce `'chardet'` when the need arises for multiple encodings? Otherwise, it looks good to me.

Reply:
chardet.detect returns the most likely encoding in the buffer. i just imported using the named export out of personal preference